File: ag.dec/agent/medicine_context.py
class MedicineContextChecker:

    # Prototype interaction data.
    # Only established examples should be added here.
    KNOWN_INTERACTIONS = {
        "warfarin": ["vitamin k rich foods"],
        "levothyroxine": ["calcium", "iron"],
    }

    def check(self, medicines, ingredients):
        medicines_lower = [medicine.lower() for medicine in medicines]
        ingredients_lower = [ingredient.lower() for ingredient in ingredients]

        relevant_checks = []
        possible_interactions = []

        for medicine in medicines_lower:

            if medicine in self.KNOWN_INTERACTIONS:
                relevant_checks.append(medicine)

                for food_item in self.KNOWN_INTERACTIONS[medicine]:

                    for ingredient in ingredients_lower:
                        if food_item in ingredient:
                            possible_interactions.append({
                                "medicine": medicine,
                                "food_component": food_item,
                                "message": (
                                    f"A known food-medicine interaction "
                                    f"may be relevant for {medicine} and "
                                    f"{food_item}."
                                )
                            })

        return {
            "medicine_checks_relevant": len(relevant_checks) > 0,
            "medicines_checked": relevant_checks,
            "possible_interactions": possible_interactions
        }

File: src/agent/medicine-context.ts
export interface MedicineResult {
  medicineChecksRelevant: boolean;
  medicinesChecked: string[];
  possibleInteractions: Array<{
    medicine: string;
    foodComponent: string;
    message: string;
  }>;
}

// Prototype interaction data — only established examples
const KNOWN_INTERACTIONS: Record<string, string[]> = {
  warfarin: ['vitamin k', 'spinach', 'kale', 'broccoli', 'collard'],
  levothyroxine: ['calcium', 'iron'],
};

export function checkMedicineContext(medicines: string[], ingredients: string[]): MedicineResult {
  const medicinesLower = medicines.map((m) => m.toLowerCase().replace(/\s+/g, ' ').trim());
  const ingredientsText = ingredients.join(' ').toLowerCase().replace(/\s+/g, ' ');

  const relevantChecks: string[] = [];
  const possibleInteractions: MedicineResult['possibleInteractions'] = [];

  for (const medicine of medicinesLower) {
    if (medicine in KNOWN_INTERACTIONS) {
      relevantChecks.push(medicine);

      for (const foodItem of KNOWN_INTERACTIONS[medicine]) {
        if (ingredientsText.includes(foodItem)) {
          possibleInteractions.push({
            medicine,
            foodComponent: foodItem,
            message: `A known food-medicine interaction may be relevant for ${medicine} and ${foodItem}.`,
          });
        }
      }
    }
  }

  return {
    medicineChecksRelevant: relevantChecks.length > 0,
    medicinesChecked: relevantChecks,
    possibleInteractions,
  };
}
